server/server.js

```diff
@@ -355,12 +361,43 @@ app.get(`${apiBaseUrl}/employer-sites-detail`,
     return res.json({ data: '' });
   }));
 
+app.post(`${apiBaseUrl}/update-user`,
+  keycloak.allowRolesMiddleware('ministry_of_health'),
+  keycloak.getUserInfoMiddleware(),
+  asyncMiddleware(async (req, res) => {
+    await validate(AccessRequestApproval, req.body);
+    await keycloak.setUserRoles(req.body.userId, req.body.role, req.body.regions);
+    await dbClient.db[collections.USERS].updateDoc(
+      {
+        keycloakId: req.body.userId,
+      },
+      {
+        sites: req.body.sites,
+      },
+    );
+
+    const user = req.hcapUserInfo;
+    logger.info({
+      action: 'update-user_post',
+      performed_by: {
+        username: user.username,
+        id: user.id,
+      },
+      role_assigned: req.body.role,
+      granted_access_to: req.body.userId,
```

Comment:
this key should probably have a different name for this type of action - user_id would be fine IMO (honestly I think it should have been like this for the other endpoint too, but that ship has sailed)

Reply:
I guess we might want to leave it as-is, even though I think it's a bad key, so that it's consistent